scitt/create_detatched_signed_statement.py

```diff
@@ -0,0 +1,217 @@
+""" Module for creating a SCITT signed statement with a detached payload"""
+
+import hashlib
+import argparse
+
+from typing import Optional
+
+from hashlib import sha256
+
+from pycose.messages import Sign1Message
+from pycose.headers import Algorithm, KID, ContentType
+from pycose.algorithms import Es256
+from pycose.keys.curves import P256
+from pycose.keys.keyparam import KpKty, EC2KpD, EC2KpX, EC2KpY, KpKeyOps, EC2KpCurve
+from pycose.keys.keytype import KtyEC2
+from pycose.keys.keyops import SignOp, VerifyOp
+from pycose.keys import CoseKey
+
+from ecdsa import SigningKey, VerifyingKey
+
+
+# Feed header label comes from version 2 of the scitt architecture document
+# https://www.ietf.org/archive/id/draft-birkholz-scitt-architecture-02.html#name-envelope-and-claim-format
+HEADER_LABEL_FEED = 392
+
+# CWT header label comes from version 4 of the scitt architecture document
+# https://www.ietf.org/archive/id/draft-ietf-scitt-architecture-04.html#name-issuer-identity
+HEADER_LABEL_CWT = 13
+
+# Various CWT header labels come from:
+# https://www.rfc-editor.org/rfc/rfc8392.html#section-3.1
+HEADER_LABEL_CWT_ISSUER = 1
+HEADER_LABEL_CWT_SUBJECT = 2
+
+# CWT CNF header labels come from:
+# https://datatracker.ietf.org/doc/html/rfc8747#name-confirmation-claim
+HEADER_LABEL_CWT_CNF = 8
+HEADER_LABEL_CNF_COSE_KEY = 1
+
+
+# Signed Hash envelope header labels from:
+# https://github.com/OR13/draft-steele-cose-hash-envelope/blob/main/draft-steele-cose-hash-envelope.md
+HEADER_LABEL_PAYLOAD_HASH_ALGORITHM = 998
+HEADER_LABEL_LOCATION_HINT = 999
+
+
+def open_signing_key(key_file: str) -> SigningKey:
+    """
+    opens the signing key from the key file.
+    NOTE: the signing key is expected to be a P-256 ecdsa key in PEM format.
```

Comment:
While this script uses P-256 ecdsa, we should list we support anything supported through [go-cose](https://github.com/veraison/go-cose)
```suggestion
    NOTE: the signing key is expected to be a P-256 ecdsa key in PEM format. 
    While this sample script uses P-256 ecdsa, DataTrails supports any format supported through [go-cose](https://github.com/veraison/go-cose/blob/main/algorithm.go)
```